### PR TITLE
Documentation/52-update-doc

### LINKED ISSUE #52: Improve documentation
Below are some important information to add to the documentation:
- [x] State minimum python version
- [x] Add quickstart instructions for the different options (i.e., Telegram, console) to run the demo
- [x] Add user and agent dialogue acts
- [x] Add recommendation dialogue flow

### PR DESCRIPTION
**Pull request by NoB0**
- Add agent and user intents
- Quickstart command in the console and Telegram.
- Add python version badge

Fixes #52

### CODE REVIEW

**Comment by mgskjaeveland** [comment]
Running quickstart for console gives me following output:
```
$ python3 -m moviebot.run -c config/moviebot_config.yaml
Running
['/home/martige/repo/misc/moviebot/moviebot/run.py', '-c', 'config/moviebot_config.yaml']
Configuration file "config/moviebot_config.yaml" is loaded.
Loading the database......
Traceback (most recent call last):
  File "/usr/lib/python3.10/runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/lib/python3.10/runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "/home/martige/repo/misc/moviebot/moviebot/run.py", line 112, in <module>
    CONTROLLER.execute_agent(CONFIGURATION)
  File "/home/martige/repo/misc/moviebot/moviebot/controller/controller_terminal.py", line 27, in execute_agent
    agent.initialize()
  File "/home/martige/repo/misc/moviebot/moviebot/agent/agent.py", line 117, in initialize
    self.nlu = NLU(data_config)
  File "/home/martige/repo/misc/moviebot/moviebot/nlu/nlu.py", line 38, in __init__
    self.intents_checker = UserIntentsChecker(config)
  File "/home/martige/repo/misc/moviebot/moviebot/nlu/user_intents_checker.py", line 42, in __init__
    self.slot_values = self.data_loader.load_database()
  File "/home/martige/repo/misc/moviebot/moviebot/nlu/data_loader.py", line 117, in load_database
    {value: self.lemmatize_value(value)})
  File "/home/martige/repo/misc/moviebot/moviebot/nlu/user_intents_checker.py", line 107, in _lemmatize_value
    [self.lemmatizer.lemmatize(word) for word in word_tokenize(value)])
  File "/home/martige/.local/lib/python3.10/site-packages/nltk/tokenize/__init__.py", line 129, in word_tokenize
    sentences = [text] if preserve_line else sent_tokenize(text, language)
  File "/home/martige/.local/lib/python3.10/site-packages/nltk/tokenize/__init__.py", line 106, in sent_tokenize
    tokenizer = load(f"tokenizers/punkt/{language}.pickle")
  File "/home/martige/.local/lib/python3.10/site-packages/nltk/data.py", line 750, in load
    opened_resource = _open(resource_url)
  File "/home/martige/.local/lib/python3.10/site-packages/nltk/data.py", line 876, in _open
    return find(path_, path + [""]).open()
  File "/home/martige/.local/lib/python3.10/site-packages/nltk/data.py", line 583, in find
    raise LookupError(resource_not_found)
LookupError: 
**********************************************************************
  Resource punkt not found.
  Please use the NLTK Downloader to obtain the resource:

  >>> import nltk
  >>> nltk.download('punkt')
  
  For more information see: https://www.nltk.org/data.html

  Attempted to load tokenizers/punkt/PY3/english.pickle

  Searched in:
    - '/home/martige/nltk_data'
    - '/usr/nltk_data'
    - '/usr/share/nltk_data'
    - '/usr/lib/nltk_data'
    - '/usr/share/nltk_data'
    - '/usr/local/share/nltk_data'
    - '/usr/lib/nltk_data'
    - '/usr/local/lib/nltk_data'
    - ''
**********************************************************************
```

I am not sure how to proceed.

File: docs/source/installation.rst [hunk 1]
```diff
@@ -28,6 +18,22 @@ nltk Resources
 
 .. code-block:: python
 
+    import nltk
```
[the remark this answers is not in the document]
Comment (NoB0) [reply docs/source/installation.rst:21]: The download of nltk resources is required to run the MovieBot. I added a line to mention that.

File: docs/source/usage.rst [hunk 2]
```diff
@@ -2,11 +2,56 @@ Usage
...
 .. code-block:: shell
 
    $ python -m moviebot.run -c <path_to_config.yaml>
        
 
-Note: To create Slot-Values, execute the code once by setting `BOT: False` in the configuration file.
\ No newline at end of file
+Note: To create Slot-Values, execute the code once by setting `BOT: False` in the configuration file.
+
+Quickstart running IAI MovieBot in the console
```
[the remark this answers is not in the document]
Comment (NoB0) [reply docs/source/usage.rst:14]: Done.

File: docs/source/installation.rst [hunk 3]
```diff
@@ -1,17 +1,7 @@
...
-- To use Telegram, one must install the Telegram application available `here <https://telegram.org/>`_.
-- Click `here <https://core.telegram.org/bots#6-botfather>`_ for instructions about how to create a Telegram Bot.
-- Add the token of the new bot to the `Telegram Bot Token <config/bot_token.yaml>`_ file as ``BOT_TOKEN: <<token>>``.
-
-Conversation History folder
----------------------------
-
-Create a folder named `conversation_history`. The conversational logs will be saved in this folder if allowed in the config file. These logs will be created when the IAI MovieBot is being executed on Telegram.
-
+The IAI MovieBot can run in the console and on Telegram.
```
[the remark this answers is not in the document]
Comment (NoB0) [reply docs/source/installation.rst:4]: Added note on the installation requirements for console and Telegram.

File: docs/source/usage.rst [hunk 4]
```diff
@@ -2,11 +2,56 @@ Usage
...
+--------------------
 
 .. code-block:: shell
 
    $ python -m moviebot.run -c <path_to_config.yaml>
        
 
-Note: To create Slot-Values, execute the code once by setting `BOT: False` in the configuration file.
\ No newline at end of file
+Note: To create Slot-Values, execute the code once by setting `BOT: False` in the configuration file.
```
[the remark this answers is not in the document]
Comment (NoB0) [reply docs/source/usage.rst:12]: Not that I am aware of.

File: docs/source/usage.rst [hunk 5]
```diff
@@ -2,11 +2,56 @@ Usage
...
+
+.. code-block:: shell
+
+   $ python -m moviebot.run -c config/moviebot_config.yaml
+
+Quickstart running IAI MovieBot with Telegram
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+To run MovieBot with Telegram you need to create a configuration file as below.
+Then, you can run the command to start the MovieBot with the new configuration.
```
[the remark this answers is not in the document]
Comment (NoB0) [reply docs/source/usage.rst]: Rephrased.

File: README.md [hunk 6]
```diff
@@ -4,6 +4,7 @@
 [![Documentation Status](https://readthedocs.org/projects/iai-moviebot/badge/?version=latest)](https://iai-moviebot.readthedocs.io/en/latest/?badge=latest)
 ![Tests](https://img.shields.io/github/workflow/status/iai-group/moviebot/Deploy?label=Tests)
```
[the remark this answers is not in the document]
Comment (NoB0) [reply README.md]: Fixed.

**Comment by NoB0** [comment]
The error is related to missing nltk resources. Please check if updates in `docs/source/installation.rst` help fix the issue.

File: docs/source/installation.rst (hunk 1 above)
[the remark this answers is not in the document]
Comment (mgskjaeveland) [reply docs/source/installation.rst:26]: Thanks! That made the difference for me.

File: docs/source/dialogue.rst [hunk 7]
```diff
@@ -8,6 +8,71 @@ The figure below shows the dialogue flow in our system.
...
++------------------------+------------+
+| Inquire  | Once the agent has recommended an item, the user can ask further details about it.        | 
++------------------------+------------+
+| Remove preference  | The user wants to remove any previously stated preference.        | 
++------------------------+------------+
+| Reject  | The user either has already seen/consumed the recommended item or does not like it.        | 
++------------------------+------------+
+| Accept  | The user accepts (likes) the recommendation. This will determine the success of the system as being able to find a recommendation the user liked.        | 
++------------------------+------------+
+| Continue recommendation  | If the user likes a recommendation, they can either restart, quit or continue the process to get a similar recommendation.        |
```
Comment (IKostric) [inline docs/source/dialogue.rst:32]: To my understanding there are different intents for restart and quit. This one is specifically a request for a new recommendation (e.g., *Recommend me something similar*).

File: docs/source/dialogue.rst [hunk 8]
```diff
@@ -8,6 +8,71 @@ The figure below shows the dialogue flow in our system.
...
++------------------------+------------+
+| Recommend  | Based on the user's preferences, make a recommendation.        | 
++------------------------+------------+
+| No results  | The database does not contain any items matching the user's preferences.        | 
++------------------------+------------+
+|  Count results | The number of items matching the user's preferences is larger than a maximum limit. This will be followed by an elicit intent.        | 
++------------------------+------------+
+| Inform  | If the user inquires about the recommended item, the agent provides the relevant information.        | 
++------------------------+------------+
+| Continue recommendation  | If the user likes a recommendation, they can either restart, quit or continue the process to get a similar recommendation.        |
```
Comment (IKostric) [inline docs/source/dialogue.rst:63]: See comment above

File: docs/source/dialogue.rst [hunk 9]
```diff
@@ -8,6 +8,71 @@ The figure below shows the dialogue flow in our system.
...
++------------------------+------------+
+| No results  | The database does not contain any items matching the user's preferences.        | 
++------------------------+------------+
+|  Count results | The number of items matching the user's preferences is larger than a maximum limit. This will be followed by an elicit intent.        | 
++------------------------+------------+
+| Inform  | If the user inquires about the recommended item, the agent provides the relevant information.        | 
++------------------------+------------+
+| Continue recommendation  | If the user likes a recommendation, they can either restart, quit or continue the process to get a similar recommendation.        | 
++------------------------+------------+
+| Restart  | The user wants to restart the recommendation process.        |
```
Comment (IKostric) [inline docs/source/dialogue.rst:65]: Description should focus on agents intent. Not sure it is actually used, as the conversation simply restarts.

File: docs/source/usage.rst [hunk 10]
```diff
@@ -2,11 +2,60 @@ Usage
...
+Note: To create Slot-Values, execute the code once by setting `BOT: False` in the configuration file.
+
+Quickstart IAI MovieBot in the console
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+To run MovieBot with default configuration execute the command below.
+
+.. code-block:: shell
+
+   $ python -m moviebot.run -c config/moviebot_config.yaml
```
Comment (IKostric) [inline docs/source/usage.rst:25]: Nit: `-c config/moviebot_config.yaml` is not needed to load the default config.

**Review by IKostric — APPROVED** [review]
LGTM. Some minor comments for the intent explanations.

File: docs/source/dialogue.rst (hunk 7 above)
Comment (NoB0) [reply docs/source/dialogue.rst:32]: After looking at the code, I see that the NLG method `generate_output` calls [_user_options_continue](https://github.com/iai-group/MovieBot/blob/bf39d971d7dffb1c3f0532dd416e577abb72a2c6/moviebot/nlg/nlg.py#L457) to propose several options, i.e, restart, quit, or similar recommendation.

File: docs/source/usage.rst (hunk 10 above)
Comment (NoB0) [reply docs/source/usage.rst:25]: Done.

**Pull request merged**